gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp

```diff
@@ -0,0 +1,508 @@
+      YAML::Node & yaml_node, const std::string& prefix)
+    {
+      std::function<void(YAML::Node, const std::string &,
+        std::shared_ptr<rclcpp_lifecycle::LifecycleNode>, const std::string& )>
+      feed_yaml_to_node_rec =
+        [&](YAML::Node yaml_node, const std::string & key,
+          std::shared_ptr<rclcpp_lifecycle::LifecycleNode> node, const std::string& prefix)
+        {
+          if (node->get_name() != prefix)
+            return;
+          static constexpr char separator = '.';
+          if (yaml_node.Type() == YAML::NodeType::Scalar) {
+            std::string val_str = yaml_node.as<std::string>();
+            //const char* val_str = yaml_node.as<const char*>();
+            // std::string val_str;
+            // YAML::convert<std::string>::decode(yaml_node, val_str);
+
+            // TODO(ddengster): Do stricter typing for set_parameter value types.
+            // (ie. Numbers should be converted to int/double/etc instead of strings)
+            //RCLCPP_ERROR(rclcpp::get_logger("load"),"key: %s val: %s", key.c_str(), val_str.c_str());
+            if (!node->has_parameter(key))
+              node->declare_parameter(key);
+
+            auto is_number = [](const std::string &str) -> bool {
+              return str.find_first_not_of("0123456789.-") == std::string::npos;
+              //@note: bugs with .05 or 15.
+            };
+
+            if (is_number(val_str))
+            {
+              std::stringstream ss(val_str);
+              double v = 0.0;
+              ss >> v;
+              node->set_parameter(rclcpp::Parameter(key, v));
+            }
+            else {
+              node->set_parameter(rclcpp::Parameter(key, val_str));
+            }
+
+            return;
+          } else if (yaml_node.Type() == YAML::NodeType::Map) {
+            for (auto yaml_node_it : yaml_node) {
+              std::string newkey = yaml_node_it.first.as<std::string>();
+              //RCLCPP_ERROR(rclcpp::get_logger("load"), "newkey: %s", newkey.c_str());
+              if (newkey == prefix || newkey == "ros__parameters")
+                newkey = "";
+              else if (!key.empty())
+                newkey = key + separator + newkey;
+              //RCLCPP_ERROR(rclcpp::get_logger("load"),"newkey (map): %s", newkey.c_str());
+              feed_yaml_to_node_rec(yaml_node_it.second, newkey, node, prefix);
+            }
+          } else if (yaml_node.Type() == YAML::NodeType::Sequence) {
+            auto it = yaml_node.begin();
+            if (yaml_node.size())
+            {
+              if (it->IsScalar())
+              {
+                //submit as array of parameters
+                std::vector<std::string> val;
+                for (auto yaml_node_it : yaml_node) {
+                  std::string name = yaml_node_it.as<std::string>();
+                  val.push_back(name);
+                  //RCLCPP_ERROR(rclcpp::get_logger("load"),"n: %s", name.c_str());
```

Comment:
please don't add commented code, the `#if` disabled macros are clearly marks as todo but these are just dangling